BUG: contentTypes Generation Fails Sliently

### Node Version

22.14.0

### NPM/Yarn/PNPM Version

PNPM 9.9.0

### Strapi Version

5.12.3

### Operating System

Windows 11

### Database

PostgreSQL

### Javascript or Typescript

Typescript

### Reproduction URL

_No response_

### Bug Description

In development using `strapi develop`, the `types/generated/contentTypes/d.ts` file silently stops updating once we have a field with negative integer `min` defined in any of the `schema.json`

Which will leading to
`error TS2345: Argument of type '"api::{{singularName}}.{{singularName}}"' is not assignable to parameter of type 'ContentType'.` when creating new content types.

### Steps to Reproduce

0. `strapi develop`
1. Open admin panel, select any content-type
2. Create a integer field
3. Set min / max range
4. Save
5. Open Terminal & Run: `pn strapi ts:generate-types`
6. Check terminal for error message
> [ERROR] [DATETIME] (Typegen) Failed to generate types for contentTypes: Debug Failure. False expression: Negative numbers should be created in combination with createPrefixUnaryExpression. Exiting

### Expected Behavior

1. Open admin panel, select any content-type
2. Create a integer field
3. Set min / max range
4. Save
5. `contentTypes.d.ts` get correctly updated

### Logs

```shell
> strapi "ts:generate-types"

[INFO]  [2025-05-03T07:55:30.172Z] (Typegen) Starting the type generation process
[INFO]  [2025-05-03T07:55:30.173Z] (Typegen) Generating types for contentTypes
[ERROR] [2025-05-03T07:55:30.178Z] (Typegen) Failed to generate types for contentTypes: Debug Failure. False expression: Negative numbers should be created in combination with createPrefixUnaryExpression. Exiting
[INFO]  [2025-05-03T07:55:30.179Z] (Typegen) The task errored with 0 warning(s) and 1 error(s) in 0.007s.
[2025-05-03 15:55:30.179] info: Shutting down Strapi
[2025-05-03 15:55:30.182] info: Strapi has been shut down
```

### Code Snippets

src/api/{{COLLECTION_NAME}}/content-types/{{COLLECTION_NAME}}/schema.json
```
    "range": {
      "type": "integer",
      "unique": false,
      "min": -12,
      "max": 20
    }
```

### Media

_No response_

### Additional information

_No response_

### Confirmation Checklist

- [x] I have checked the existing [issues](https://github.com/strapi/strapi/issues) for duplicates.
- [x] I agree to follow this project's [Code of Conduct](https://github.com/strapi/strapi/blob/develop/CODE_OF_CONDUCT.md).

I have exactly same error, happens from nowhere and have no idea whats causing it.
its just not updating contentTypes.d.ts which causes errors.

I found that issue ticket by debuging same way so after try `npx strapi ts:generate-types`:
`[INFO]  [2025-05-09T12:23:23.898Z] (Typegen) Starting the type generation process
[DEBUG] [2025-05-09T12:23:23.898Z] (Typegen) Enabled artifacts: contentTypes, components
[INFO]  [2025-05-09T12:23:23.899Z] (Typegen) Generating types for contentTypes
[DEBUG] [2025-05-09T12:23:23.907Z] (Typegen) Found 36 content-types.
[DEBUG] [2025-05-09T12:23:24.173Z] (Typegen) Generated contentTypes in 0.274s
[INFO]  [2025-05-09T12:23:24.174Z] (Typegen) Generating types for components
[ERROR] [2025-05-09T12:23:24.175Z] (Typegen) Failed to generate types for components: Debug Failure. False expression: Negative numbers should be created in combination with createPrefixUnaryExpression. Exiting
[INFO]  [2025-05-09T12:23:24.176Z] (Typegen) The task errored with 0 warning(s) and 1 error(s) in 0.278s.`

i tried to add collection with content-builder and add there just one text input or whatever else, trying to find solution

--- a/packages/utils/typescript/lib/__tests__/generators/schemas/utils.test.js
+++ b/packages/utils/typescript/lib/__tests__/generators/schemas/utils.test.js
@@ -168,10 +168,16 @@ describe('Utils', () => {
     });
 
     test('Number', () => {
-      const node = toTypeLiteral(42);
+      const nodePositive = toTypeLiteral(42);
+      const nodeNegative = toTypeLiteral(-42);
 
-      expect(node.kind).toBe(ts.SyntaxKind.FirstLiteralToken);
-      expect(node.text).toBe('42');
+      expect(nodePositive.kind).toBe(ts.SyntaxKind.FirstLiteralToken);
+      expect(nodePositive.text).toBe('42');
+
+      expect(nodeNegative.kind).toBe(ts.SyntaxKind.PrefixUnaryExpression);
+      expect(nodeNegative.operator).toBe(ts.SyntaxKind.MinusToken);
+      expect(nodeNegative.operand.kind).toBe(ts.SyntaxKind.FirstLiteralToken);
+      expect(nodeNegative.operand.text).toBe('42');
     });
 
     test('Boolean', () => {

--- a/packages/utils/typescript/lib/generators/common/models/utils.js
+++ b/packages/utils/typescript/lib/generators/common/models/utils.js
@@ -92,7 +92,12 @@ const toTypeLiteral = (data) => {
   }
 
   if (isNumber(data)) {
-    return factory.createNumericLiteral(data);
+    return data < 0
+      ? factory.createPrefixUnaryExpression(
+          ts.SyntaxKind.MinusToken,
+          factory.createNumericLiteral(-data)
+        )
+      : factory.createNumericLiteral(data);
   }
 
   if (isBoolean(data)) {

